qemu/tests/vdpa_pxe_boot.py

```diff
@@ -0,0 +1,24 @@
+import time
+
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    vdpa nic PXE test:
+
+    1) Boot up guest from vdpa NIC
+    2) Check guest if works well after sleep 240s
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env: Dictionary with test environment.
+    """
+    error_context.context("Try to boot from vdpa NIC", test.log.info)
+    vm = env.get_vm(params["main_vm"])
+    timeout = params.get_numeric("pxe_timeout")
+    test.log.info("Waiting %ss" % timeout)
+    time.sleep(timeout)
+    vm.verify_alive()
```

Comment:
> I cannot see any checkpoint regarding pxe, such as tcpdump, check guest serial output, or others. The `vm.verify_alive()` is not enough to check if the guest is boot from pxe I think.
Hi @PaulYuuu First of all, thank you for your help in reviewing my patch. Because vdpa nic cannot connect to the pxe server temporarily, I did not add the `tcpdump` step. In addition, according to past test results, at some point, using vdpa nic pxe guest will cause core dump, so I try to capture the guest status using `vm.verify_alive()`. Please correct me if my design was wrong, thanks in advance.

Reply:
OK, so the vdpa nic is isolated, the guest can still boot from PXE, but the host cannot capture any packages, right?

If so, you should check the guest console output to ensure it really boots from PXE without any errors.

Because without PXE server support, this case is boot from empty image.

Reply:
Thanks for the suggestion, I'll add a steps check qemu output in my next update.